src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -345,24 +344,20 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams, boolean v1
 
         MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
 
-        if (V_2.equals(callbackParams.getVersion()) && caseData.hasApplicantRejectedRepaymentPlan()) {
-            response.state(CaseState.PROCEEDS_IN_HERITAGE_SYSTEM.name());
-        } else {
-            if (v1 && featureToggleService.isSdoEnabled()) {
-                if (caseData.getRespondent1ClaimResponseTypeForSpec().equals(RespondentResponseTypeSpec.FULL_DEFENCE)) {
-                    if ((multiPartyScenario.equals(ONE_V_ONE) || multiPartyScenario.equals(TWO_V_ONE))
-                        || multiPartyScenario.equals(ONE_V_TWO_ONE_LEGAL_REP)) {
-                        response.state(CaseState.JUDICIAL_REFERRAL.name());
-                    } else if (multiPartyScenario.equals(ONE_V_TWO_TWO_LEGAL_REP)) {
-                        if (caseData.getRespondent2ClaimResponseTypeForSpec()
-                            .equals(RespondentResponseTypeSpec.FULL_DEFENCE)) {
-                            response.state(CaseState.JUDICIAL_REFERRAL.name());
-                        }
-                    }
-                }
-            }
+        if (v1 && featureToggleService.isSdoEnabled()) {
+            putCaseStateInJudicialReferral(caseData, response);
         }
 
+        if (V_2.equals(callbackParams.getVersion()) && featureToggleService.isPinInPostEnabled()) {
+            if (isOneVOne(caseData)
```

Comment:
perhaps remove stuff to do with in mediation?

Reply:
After merging with master, this problem should be resolved